### PR TITLE
SDI-381 move / add to hospital validation

### PR DESCRIPTION
**Pull request by mikehalmamoj**

### CODE REVIEW

File: server/routes/addPatient/prisonerSelect.ts [hunk 1]
```diff
@@ -35,6 +35,8 @@ export default class PrisonerSelectRoutes {
     const missingPatients = searchResults
       .filter(result => !result.restrictedPatient)
       .filter(result => result.lastMovementTypeCode === 'REL' && result.lastMovementReasonCode === 'HP')
+      .filter(result => result.indeterminateSentence || result.conditionalReleaseDate > new Date())
```
Comment (petergphillips) [inline server/routes/addPatient/prisonerSelect.ts:38]: got a feeling here you should be excluding recalls

File: server/routes/movePrisoner/prisonerSelect.ts [hunk 2]
```diff
@@ -33,7 +33,11 @@ export default class PrisonerSelectRoutes {
       user
     )
 
-    return this.renderView(req, res, { searchResults, searchTerm })
+    const availablePrisoners = searchResults
+      .filter(result => result.indeterminateSentence || result.conditionalReleaseDate > new Date())
```
Comment (petergphillips) [inline server/routes/movePrisoner/prisonerSelect.ts:37]: bit naff to have logic in two places - could you extract out filter function and re-use?

Comment (mikehalmamoj) [reply server/routes/movePrisoner/prisonerSelect.ts:37]: I was thinking Martin Fowler's rule of 3 😁. But fair enough, probably better to have their own tests anyway.

File: server/routes/searchPatients/restrictedPatientSearchFilter.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  private releasedToHospital = (prisoner: PrisonerSearchSummary): boolean => {
+    return prisoner.lastMovementTypeCode === 'REL' && prisoner.lastMovementReasonCode === 'HP'
+  }
+
+  private determinateSentenceAfterCRD = (prisoner: PrisonerSearchSummary): boolean => {
+    return !prisoner.recall && !prisoner.indeterminateSentence && prisoner.conditionalReleaseDate < new Date()
+  }
+
+  private recallAfterSED = (prisoner: PrisonerSearchSummary): boolean => {
+    return prisoner.recall && prisoner.sentenceExpiryDate < new Date()
```
Comment (petergphillips) [inline server/routes/searchPatients/restrictedPatientSearchFilter.ts:39]: I worried that you should also be excluding `indeterminateSentence` here, but can't find any prisoners on a recall with a `sentenceExpiryDate` set anyway so think we're good

**Review by petergphillips — APPROVED** [review]
LGTM

**Pull request merged**